greenriver/hmis-frontend
# Stricter behavior for form property editor

## Description

Update property editor to be match stricter schema. Sorry this is a big diff here. It includes:
* Only show the "relevant" fields in EnableWhen/AUtofill/Initial sections
* Some bug fixes on formUtil
* Some UI improvements

Relates to (but does not depend on): https://github.com/greenriver/hmis-warehouse/pull/4498 which enforces the stricter schema


**How to test**:
* pull down the related backend branch `stricter-hmis-form-json-schema`
* you will probalby need to delete any local forms that you have updated using the form builder to date, because it injected values that are now considered invalid
* use the form builder to create new forms. they should be valid (Eg not a bunch of JSON Schema errors when you save)

## Type of change

## Checklist before requesting review
